main.js

```diff
@@ -41,56 +47,113 @@ const drawBall = () => {
 };
 
 const updatePaddles = () => {
-  console.log(paddleOneSpeed);
-
-  paddleOneSpeed -= paddleOneSpeed * friction;
-
-  if (paddleOneSpeed > maxSpeed) {
-    paddleOneSpeed = maxSpeed;
-  }
-  if (paddleOneSpeed < -maxSpeed) {
-    paddleOneSpeed = -maxSpeed;
-  }
-  if (paddleTwoSpeed > maxSpeed) {
-    paddleTwoSpeed = maxSpeed;
-  }
-  if (paddleTwoSpeed < -maxSpeed) {
-    paddleTwoSpeed = -maxSpeed;
-  }
+  paddleOneSpeed -= friction * paddleOneSpeed;
+  paddleTwoSpeed -= friction * paddleTwoSpeed;
 
+  //if they are moving pretty slow, just stop them.
   if (Math.abs(paddleOneSpeed) < friction) {
     paddleOneSpeed = 0;
   }
-
   if (Math.abs(paddleTwoSpeed) < friction) {
     paddleTwoSpeed = 0;
   }
 
+  //add intertia from keypresses
+  if (paddleOneControls.up) {
+    paddleOneSpeed -= acceleration;
+  }
+  if (paddleOneControls.down) {
+    paddleOneSpeed += acceleration;
+  }
+  if (paddleTwoControls.up) {
+    paddleTwoSpeed -= acceleration;
+  }
+  if (paddleTwoControls.down) {
+    paddleTwoSpeed += acceleration;
+  }
+
   paddleOne.y += paddleOneSpeed;
   paddleTwo.y += paddleTwoSpeed;
 
+  //bounce off the top and bottom
   if (paddleOne.y < 0) {
     paddleOne.y = 0;
+    paddleOneSpeed *= -1;
   }
   if (paddleOne.y > canvas.height - paddleOne.height) {
     paddleOne.y = canvas.height - paddleOne.height;
+    paddleOneSpeed *= -1;
   }
   if (paddleTwo.y < 0) {
     paddleTwo.y = 0;
+    paddleTwoSpeed *= -1;
   }
   if (paddleTwo.y > canvas.height - paddleTwo.height) {
     paddleTwo.y = canvas.height - paddleTwo.height;
+    paddleTwoSpeed *= -1;
   }
 };
 
 const updateBall = () => {
+  if (justHit) {
+    if (ball.x < buffer) {
```

Comment:
i don't get what buffer is doing here

Reply:
buffer and justHit mean that if the ball has just changed direction, it will NOT change direction again until it moves outside of the "buffer" zone at each end of the canvas width.  This prevents it from getting caught in/on/behind the paddle or outer border of the canvas.